api/scpca_portal/management/commands/load_data.py

```diff
@@ -88,77 +100,161 @@ def project_has_s3_files(project_id: str) -> bool:
-
         # Prepare data input directory.
         common.INPUT_DATA_PATH.mkdir(exist_ok=True, parents=True)
 
         # Prepare data output directory.
         shutil.rmtree(common.OUTPUT_DATA_PATH, ignore_errors=True)
         common.OUTPUT_DATA_PATH.mkdir(exist_ok=True, parents=True)
 
-        allowed_submitters = allowed_submitters or ALLOWED_SUBMITTERS
-        project_id = kwargs.get("scpca_project_id")
-
-        s3.download_input_metadata()
+        s3.download_input_metadata(kwargs["input_bucket_name"])
 
-        project_list = metadata_file.load_projects_metadata(
+        projects_metadata = metadata_file.load_projects_metadata(
             Project.get_input_project_metadata_file_path()
         )
-        for project_data in project_list:
-            scpca_project_id = project_data["scpca_project_id"]
-            if project_id and project_id != scpca_project_id:
-                continue
+        for project_metadata in projects_metadata:
+            metadata_project_id = project_metadata["scpca_project_id"]
+            passed_project_id = kwargs.get("scpca_project_id")
 
-            if not self.project_has_s3_files(scpca_project_id):
-                logger.warning(
-                    f"Metadata found for '{scpca_project_id}' but no s3 folder of that name exists."
-                )
-                return
-
-            if project_data["pi_name"] not in allowed_submitters:
-                logger.warning("Project submitter is not the white list.")
+            if self.skip_project(
+                metadata_project_id, passed_project_id, project_metadata["pi_name"]
+            ):
                 continue
 
-            if project := Project.objects.filter(scpca_id=scpca_project_id).first():
-                # Purge existing projects so they can be re-added.
-                if kwargs["reload_all"] or kwargs["reload_existing"]:
-                    logger.info(f"Purging '{project}")
-                    project.purge(delete_from_s3=kwargs["update_s3"])
-                # Only import new projects.
-                # If old ones are desired they should be purged and re-added.
-                else:
-                    logger.info(f"'{project}' already exists. Use --reload-existing to re-import.")
+            # If project exists and cannot be purged, then throw a warning
+            if project := Project.objects.filter(scpca_id=metadata_project_id).first():
+                purge_project_kwargs = utils.filter_dict_by_keys(
+                    kwargs, {"reload_all", "reload_existing", "update_s3"}
+                )
+                if not self.purge_project(project, **purge_project_kwargs):
                     continue
 
-            logger.info(f"Importing '{project}' data")
-            project = Project.get_from_dict(project_data)
+            logger.info(f"Importing 'Project {metadata_project_id}' data")
+            project_metadata["s3_input_bucket"] = kwargs["input_bucket_name"]
```

Comment:
@davidsmejia curious to see if you have an idea as to how (and when) to assign the `s3_input_bucket` attribute to `Project` objects.

Reply:
This instead
```suggestion
            project = Project.get_from_dict(project_metadata)
            project.s3_input_bucket = kwargs["input_bucket_name"]
            project.save()
```